src/renderer/src/lib.rs

```diff
@@ -184,31 +190,99 @@ pub struct Fragment<R: gfx::Resources> {
     pub ka: Texture<R>,
     /// diffuse color
     pub kd: Texture<R>,
+    /// specular color
+    pub ks: Texture<R>,
+    /// specular exponent
+    pub ns: f32,
 }
 
-/// A basic light
+
+/// Represents a point light.
 #[derive(Copy, Clone)]
-pub struct Light {
+pub struct PointLight {
     /// The XYZ coordinate of the light
     pub center: [f32; 3],
-    /// How big the light is radius, lighting
-    /// passes are not required to render the light
-    /// beyond this radius
-    pub radius: f32,
 
     /// The color of light emitted
     pub color: [f32; 4],
-    /// constant, propagation means no falloff of the light
-    /// emission from distance. Useful for the sun.
-    pub propagation_constant: f32,
-    /// linear level drops
-    pub propagation_linear: f32,
-    /// cubic light level drop
-    pub propagation_r_square: f32,
+
+    /// Represents constant, linear, and quadratic (kc, kl, and kq)
+    /// factors for light attenuation calculations. See these links
+    /// for some more information on how to choose good values for
+    /// these factors:
+    /// https://imdoingitwrong.wordpress.com/2011/01/31/light-attenuation/
+    /// http://ogldev.atspace.co.uk/www/tutorial20/tutorial20.html
+    pub attenuation: [f32; 3],
+}
+
+impl PointLight {
+    /// Creates a new PointLight with the attenuation factors
+    /// set to create a light that falls below a cutoff at `radius`.
+    /// Since most monitors won't display anything less than an RGB
+    /// value of 1/255, this function uses that value as a cutoff.
+    pub fn from_radius(radius: f32) -> PointLight {
```

Comment:
Good point regarding cutoff! Useful for possible light culling.
(I'm not really convinced by the display range argumentation with respect to HDR rendering.)